apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -957,28 +957,41 @@ func (m *ModelManagerAPI) ListModels(ctx context.Context, user params.Entity) (p
 		return result, errors.Trace(err)
 	}
 
-	modelInfos, err := m.state.ModelBasicInfoForUser(userTag, m.isAdmin)
+	ctrlUser, err := m.userService.GetUserByName(ctx, userTag.Name())
```

Comment:
This is what I disagree with, although maybe others are ok with it. The facade should pass down what it receives over the wire (in this case effectively user name) to the service layer so that the resulting operation can be performed atomically. Here, we are splitting a single operation across multiple service calls and breaking atomicity.

Reply:
I understand what you are saying. But we aren't and here is why.

- The authenticated user through facade is getting transformed to a UUID now. That is not atomic.
- Where it matters I have made all of the model code check the uuid exists and return the appropriate error when it doesn't. So if the rug does get pulled out from underneath us the state layer will handle it correctly. You can run the same check twice to build a better model without having to try and massage everything into what the state layer needs for one shot.

Reply:
Fair point. Thought it's not the state layer I was talking about but the service layer. And unrelated, we do still have txn boundary issues there we ultimately need to solve.
As discussed live, passing the user UUID makes sense.